### PR TITLE
Event & Participant Management

### PR DESCRIPTION
**Pull request by shreyas-satish**
- Manage events, ticket types and participants for a proposal space.
- Import participant data from external ticket vendors (currently, there's only support for Explara).
- Make HTML badges (with QR Codes) for participants.
- Enable check-in process for Participants.
- Serve API for basic contact exchange through [zalebi](https://github.com/hasgeek/zalebi).

### CODE REVIEW

File: funnel/models/contact_exchange.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+__all__ = ['ContactExchange']
+
+
+class ContactExchange(BaseMixin, db.Model):
+    __tablename__ = 'contact_exchange'
+    user_id = db.Column(None, db.ForeignKey('user.id'))
+    proposal_space_id = db.Column(None, db.ForeignKey('proposal_space.id'))
+    participant_id = db.Column(None, db.ForeignKey('participant.id'))
+    __table_args__ = (db.UniqueConstraint('user_id', 'proposal_space_id', 'participant_id'),)
```
Comment (jace) [inline funnel/models/contact_exchange.py]: Since there's nothing else in this table, this could be a three column primary key with no native `id` column.

Comment (jace) [inline funnel/models/contact_exchange.py]: Also, I can't make out what this model is for. Is this for a user receiving another participant's contact info? A docstring will help.

File: funnel/models/event.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    constitutes as an independent event.
+    This is done to allow distinguishing participants based on
+    on the tickets they have, given a participant may have a ticket
+    for only the workshop or a single day of the conference.
+    An event is associated with multiple ticket types,
+    which helps make the distinction between participants.
+    """
+    __tablename__ = 'event'
+
+    name = db.Column(db.Unicode(80), nullable=False)
```
Comment (jace) [inline funnel/models/event.py]: Is `name` here Explara-linked? We'll need a more backend-agnostic approach.

Comment (shreyas-satish) [inline funnel/models/event.py]: No, it isn't tied to the naming on Explara. Actually, nothing in this file is specific to Explara.

Comment (jace) [inline funnel/models/event.py]: Should use `BaseScopedNameMixin` with `name` and `title` then.

**Comment by shreyas-satish** [comment]
Irish Cream PR awaits.

**Comment by shreyas-satish** [comment]
This isn't ready for master yet. Some major changes coming up.

**Pull request closed without merging**